app/routes.py

```diff
@@ -56,9 +56,11 @@ def submit():
             pathString += i + "|"
     pathString += challenge["endArticle"]
     if submitForm.validate_on_submit():
+        if not checkValidPath(pathString.split("|")):
+            return render_template('view.html', form=form, submitForm=submitForm, submitted=False, challenge=challenge, article_errors=[], path=pathString, path_error=True)
         createNewSubmission(current_user.id, challenge["id"], pathString, int(time.time()))
         return redirect(url_for('view', id=int(submitForm.challenge_id.data)))
-    return render_template('view.html', form=form, submitForm=submitForm, submitted=False, challenge=challenge, errors=submitForm.errors["path"][0], path=pathString)
+    return render_template('view.html', form=form, submitForm=submitForm, submitted=False, challenge=challenge, article_errors=submitForm.errors["path"][0], path=pathString, path_error=False)
```

Comment:
path_error=Boolean I don't think is enough here. We should report back which path links are invalid, and place them under the relevant articles.

Exactly like I've done with article_errors